autoPyTorch/api/base_task.py

```diff
@@ -187,7 +189,16 @@ def __init__(
 
         self.stop_logging_server = None  # type: Optional[multiprocessing.synchronize.Event]
 
+        # Single core, local runs should use fork
+        # to prevent the __main__ requirements in
+        # examples. Nevertheless, multi-process runs
+        # have spawn as requirement to reduce the
+        # possibility of a deadlock
         self._dask_client = None
+        self._multiprocessing_context = 'forkserver'
+        if self.n_jobs == 1:
+            self._multiprocessing_context = 'fork'
+            self._dask_client = SingleThreadedClient()
```

Comment:
This single threaded client aims to do this, while still providing an interface similar to dask client.

The problem is that many places in the code assume we are working with futures (because usually we are, as usually we run with more than 1 core). If running with 1 core, we do not want to use dask but also we do not want to add an `if` statement everywhere in smac/auto-sklearn/autopytorch. So this seems like a good tradeoff.

What this gives you is no new process (single thread) with the same dask interface.

Reply:
Ok, I will check after a while and if there are some other issues, I will fix this part at this moment.